### PR TITLE
fix build errors when tracing mode = 1

### PR DESCRIPTION
**Pull request by okkwon**
This fixes the compilation error when the tracing mode is set to 1.

### CODE REVIEW

File: runtime/src/iree/hal/drivers/cuda/tracing.c [hunk 1]
```diff
@@ -6,7 +6,7 @@
 
 #include "iree/hal/drivers/cuda/tracing.h"
 
-#if IREE_TRACING_FEATURES & IREE_TRACING_FEATURE_INSTRUMENTATION
+#if IREE_TRACING_FEATURES & IREE_TRACING_FEATURE_INSTRUMENTATION_DEVICE
```
Comment (ScottTodd) [inline runtime/src/iree/hal/drivers/cuda/tracing.c:9]: Nice, I can drop these changes from https://github.com/openxla/iree/pull/16454 with this.

Do you know if these errors are still there with `-DIREE_TRACING_PROVIDER=console`? https://github.com/openxla/iree/actions/runs/7963968500/job/21740569823?pr=16454#step:4:2056

```
/work/runtime/src/iree/base/tracing/console.c:145: undefined reference to `iree_time_now'
  /usr/bin/ld: /work/runtime/src/iree/base/tracing/console.c:145: undefined reference to `iree_time_now'
```

(I didn't have a Linux system to test on)

**Review by benvanik — APPROVED** [review]

File: runtime/src/iree/hal/drivers/cuda/tracing.c (hunk 1 above)
Comment (okkwon) [reply runtime/src/iree/hal/drivers/cuda/tracing.c:9]: That particular error is not present, but there are other build errors with unused variables, which I can quickly fix.

But there is a runtime crash, but I am not sure if this is my fault not setting something for the console tracing or not.

```
[910/1101] Generating shift_ops.vmfb from shift_ops.mlir
FAILED: runtime/src/iree/vm/test/shift_ops.vmfb /usr/local/google/home/okwan/iree/build/runtime/src/iree/vm/test/shift_ops.vmfb 
cd /usr/local/google/home/okwan/iree/build/runtime/src/iree/vm/test && /usr/local/google/home/okwan/iree/build/tools/iree-compile --output-format=vm-bytecode --mlir-print-op-on-diagnostic=false --compile-mode=vm /usr/local/google/home/okwan/iree/runtime/src/iree/vm/test/shift_ops.mlir -o shift_ops.vmfb --iree-hal-executable-object-search-path=\"/usr/local/google/home/okwan/iree/build\"
Please report issues to https://github.com/openxla/iree/issues and include the crash backtrace.
Stack dump:
0.	Program arguments: /usr/local/google/home/okwan/iree/build/tools/iree-compile --output-format=vm-bytecode --mlir-print-op-on-diagnostic=false --compile-mode=vm /usr/local/google/home/okwan/iree/runtime/src/iree/vm/test/shift_ops.mlir -o shift_ops.vmfb --iree-hal-executable-object-search-path=\"/usr/local/google/home/okwan/iree/build\"
 #0 0x00007fad0f3599c8 llvm::sys::PrintStackTrace(llvm::raw_ostream&, int) (/usr/local/google/home/okwan/iree/build/lib/libIREECompiler.so+0x4b599c8)
 #1 0x00007fad0f35779e llvm::sys::RunSignalHandlers() (/usr/local/google/home/okwan/iree/build/lib/libIREECompiler.so+0x4b5779e)
 #2 0x00007fad0f35a078 SignalHandler(int) Signals.cpp:0:0
 #3 0x00007fad0a25a510 (/lib/x86_64-linux-gnu/libc.so.6+0x3c510)
 #4 0x00007fad0a27ac0e __vfprintf_internal ./stdio-common/vfprintf-internal.c:1440:3
 #5 0x00007fad0a270306 fprintf ./stdio-common/fprintf.c:36:1
 #6 0x00007fad125d2767 iree_tracing_zone_begin_impl console.c:0:0
 #7 0x00007fad0f485554 mlir::iree_compiler::IREE::VM::translateModuleToBytecode(mlir::iree_compiler::IREE::VM::ModuleOp, mlir::iree_compiler::IREE::VM::TargetOptions, mlir::iree_compiler::IREE::VM::BytecodeTargetOptions, llvm::raw_ostream&) BytecodeModuleTarget.cpp:0:0
 #8 0x00007fad0f48a046 mlir::iree_compiler::IREE::VM::translateModuleToBytecode(mlir::ModuleOp, mlir::iree_compiler::IREE::VM::TargetOptions, mlir::iree_compiler::IREE::VM::BytecodeTargetOptions, llvm::raw_ostream&) BytecodeModuleTarget.cpp:0:0
 #9 0x00007fad0f2b33c2 ireeCompilerInvocationOutputVMBytecode (/usr/local/google/home/okwan/iree/build/lib/libIREECompiler.so+0x4ab33c2)
#10 0x00007fad0f4a79a2 mlir::iree_compiler::runIreecMain(int, char**)::$_2::operator()(iree_compiler_source_t*) const iree_compile_lib.cc:0:0
#11 0x00007fad0f4a708d mlir::iree_compiler::runIreecMain(int, char**) iree_compile_lib.cc:0:0
#12 0x00007fad0a2456ca __libc_start_call_main ./csu/../sysdeps/nptl/libc_start_call_main.h:74:3
#13 0x00007fad0a245785 call_init ./csu/../csu/libc-start.c:128:20
#14 0x00007fad0a245785 __libc_start_main ./csu/../csu/libc-start.c:347:5
#15 0x000055b6ffcea6d1 _start (/usr/local/google/home/okwan/iree/build/tools/iree-compile+0x16d1)
```

Comment (ScottTodd) [reply runtime/src/iree/hal/drivers/cuda/tracing.c:9]: Ah you likely need to turn off compiler tracing (`-DIREE_ENABLE_COMPILER_TRACING=OFF`) to use 'console'.

Comment (okkwon) [reply runtime/src/iree/hal/drivers/cuda/tracing.c:9]: That is already off.

Comment (ScottTodd) [reply runtime/src/iree/hal/drivers/cuda/tracing.c:9]: ¯\\\_(ツ)\_/¯ okay, fine to ignore then. Thanks for the fixes!

**Pull request merged**